From 1adadf4517ee19010ef3fa71a2e424c33180c42c Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Wed, 8 Oct 2014 15:57:20 +0200
Subject: [PATCH 3/4] bcmdhd: Clear MMC_KEEP_POWER flag on resume

This flag is used to tell MMC/SDIO controller
to stay awake when platform goes in suspend.
It needs to be cleared upon resume.
Also, make sure the print monitoring this flag
usage is always visible in case of errors

Change-Id: Ibfc224f03579776718c6bdb64147f1c403366f42
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
---
 drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c b/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c
index f89b81bea875..a461d7316227 100644
--- a/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c
+++ b/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c
@@ -37,6 +37,7 @@
 #include <linux/mmc/host.h>
 #include <linux/mmc/sdio_func.h>
 #include <linux/mmc/sdio_ids.h>
+#include <linux/mmc/sdhci.h>
 #include <dhd_linux.h>
 #include <bcmsdh_sdmmc.h>
 #include <dhd_dbg.h>
@@ -229,7 +230,7 @@ static int bcmsdh_sdmmc_suspend(struct device *pdev)
 
 	sdio_flags = sdio_get_host_pm_caps(func);
 	if (!(sdio_flags & MMC_PM_KEEP_POWER)) {
-		sd_err(("%s: can't keep power while host is suspended\n", __FUNCTION__));
+		pr_err("%s: can't keep power while host is suspended\n", __FUNCTION__);
 		return  -EINVAL;
 	}
 
@@ -262,6 +263,8 @@ static int bcmsdh_sdmmc_resume(struct device *pdev)
 #if defined(OOB_INTR_ONLY)
 	bcmsdh_resume(sdioh->bcmsdh);
 #endif 
+	if (func->card && func->card->host)
+		func->card->host->pm_flags &= ~MMC_PM_KEEP_POWER;
 
 	smp_mb();
 	return 0;
-- 
1.7.9.5

